app/src/main/java/com/infomaniak/mail/utils/MessageUtils.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Infomaniak Mail - Android
+ * Copyright (C) 2025 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.mail.utils
+
+import android.os.Parcelable
+import com.infomaniak.mail.data.cache.mailboxContent.MessageController
+import com.infomaniak.mail.data.cache.mailboxContent.ThreadController
+import com.infomaniak.mail.data.models.mailbox.Mailbox
+import kotlinx.parcelize.Parcelize
+
+object MessageUtils {
+    fun getMessageUidToReply(
+        threadController: ThreadController,
+        messageController: MessageController,
+        featureFlagsLive: Mailbox.FeatureFlagSet?,
+        threadsUids: List<String>
+    ): List<JunkMessageThreadData> {
+        val threadList = threadController.getThreads(threadsUids)
+        val result = mutableListOf<JunkMessageThreadData>()
+        threadList.forEach { thread ->
+            result.add(
+                JunkMessageThreadData(
+                    threadUid = thread.uid,
+                    messageUid = messageController.getLastMessageToExecuteAction(thread, featureFlagsLive).uid
+                )
+            )
+        }
+
+        return result
```

Comment:
Instead of creating a variable for a list then adding, we can directly use `map` to create the list